drivers/include/isl29125.h

```diff
@@ -0,0 +1,152 @@
+ */
+
+#ifndef ISL29125_H
+#define ISL29125_H
+
+#include <stdint.h>
+
+#include "periph/i2c.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+ /**
+  * @brief The sensors hard coded I2C address
+  */
+#define ISL29125_I2C_ADDRESS 0x44
+
+/**
+ * @brief data type for storing rgb sensor readings
+ */
+typedef struct {
+    float red;              /**< red lux value */
+    float green;            /**< green lux value */
+    float blue;             /**< blue lux value */
+} isl29125_rgb_t;
+
+/**
+ * @brief supported operation modes of the ISL29125 sensor's ADC
+ * conversion
+ */
+typedef enum {
+    ISL29125_MODE_DOWN,     /**< ADC conversion powered down */
+    ISL29125_MODE_STANDBY,  /**< ADC conversion not performed */
+    ISL29125_MODE_RGB,      /**< RGB conversion */
+    ISL29125_MODE_R,        /**< red conversion only */
+    ISL29125_MODE_G,        /**< green conversion only */
+    ISL29125_MODE_B,        /**< blue conversion only */
+    ISL29125_MODE_RG,       /**< red and green conversion only */
+    ISL29125_MODE_GB,       /**< green and blue conversion only */
+} isl29125_mode_t;
+
+/**
+ * @brief supported RGB sensing range values of the ISL29125 sensor
+ */
+typedef enum {
+    ISL29125_RANGE_375,      /**< range: 5.7m - 375 lux */
+    ISL29125_RANGE_10K,      /**< range: 0.152 - 10,000 lux */
+} isl29125_range_t;
+
+/**
+ * @brief supported color resolutions of the ISL29125 sensor's ADC
+ * conversion
+ */
+typedef enum {
+    ISL29125_RESOLUTION_12,      /**< resolution: 12 bit */
+    ISL29125_RESOLUTION_16,      /**< resolution: 16 bit */
+} isl29125_resolution_t;
+
+/**
+ * @brief Device descriptor for ISL29125 sensors
+ */
+typedef struct {
+    i2c_t i2c;                      /**< I2C device the sensor is connected to */
+    uint8_t address;                /**< I2C bus address of the sensor */
+    isl29125_range_t range;         /**< sensor range */
+    isl29125_resolution_t res;      /**< sensor resolution */
+} isl29125_t;
+
+/**
+ * @brief initialize a new ISL29125 device
+ *
+ * @param[in] dev           device descriptor of an ISL29125 device
+ * @param[in] i2c           I2C device the sensor is connected to
+ * @param[in] address       I2C address of the sensor
+ * @param[in] mode          operation mode
+ * @param[in] range         measurement range
+ * @param[in] resolution    ADC conversion resolution
+ *
+ * @return              0 on success
+ * @return              -1 on error
+ */
+int isl29125_init(isl29125_t *dev, i2c_t i2c, uint8_t address,
+        isl29125_mode_t mode, isl29125_range_t range,
+        isl29125_resolution_t resolution);
+
+/**
+ * @brief read RGB values from device
+ *
+ * @param[in] dev       device descriptor of an ISL29125 device
+ * @param[in] dest      pointer to color object data is written to
+ *
+ * @return              0 on success
+ * @return              -1 on error
+ */
+int isl29125_read_rgb(isl29125_t *dev, isl29125_rgb_t *dest);
```

Comment:
I originally defined color16_t and color8_t to avoid this precision loss (still available in the corresponding wip branch), but threw it out again. If I were to re-add it, I'd rather do it in addition to the existing definition.